### PR TITLE
Windows CI example parity: CMake-native build/run helpers (Phase 1)

### PR DESCRIPTION
**Pull request by jgmelber**
## Summary

Migrates `programming_examples` to build and run via **CMake + ctest** instead of
GNU `make`, so the same examples get CI coverage on **both Linux and native
Windows**. The Windows Ryzen AI job currently skips ~126 make-gated tests
(`REQUIRES: makefile_examples`, disabled on Windows); this is the foundation for
closing that gap.

**Scope:** CMake migration only. WSL deprecation (#3453) is a separate follow-up.
Remaining example conversions, the full inventory, and the design notes are
tracked in **#3459**.

## What's in this PR

**Helpers** (`programming_examples/common.cmake`)
- `add_aie_design` — JIT a design's `xclbin`/`insts` via `add_custom_command`
  (the CMake equivalent of `makefile-common`'s `jit_xclbin`).
- `add_aie_run_test` — register a `ctest` that runs on the NPU through
  `utils/run_on_npu.py`, covering three shapes:
  `EXE` (host binary), `PY` (the `run_py` flow: `test.py --xclbin … --instr … -k`),
  and `PY_STANDALONE` (`@iron.jit` designs that build and run themselves).
- No shell involved — `$<TARGET_FILE:>` handles the `.exe` suffix on Windows, so
  no `powershell`/`wslpath` wrapping is needed on any host.

**lit**
- New `cmake_examples` feature, enabled wherever `cmake` is present (including
  Windows — unlike the make gate).

**5 examples converted** (all had `aieargs = -d ${devicename}`, so they take the
helpers unchanged): `matrix_scalar_add`, `vector_reduce_add`, `vector_reduce_min`,
`vector_scalar_add`, `vector_vector_modulo`.

Two `ml/block_datatypes` examples were also converted and then **reverted**:
appending the helpers to their `CMakeLists.txt` also changed the *make* build
(`build_host_exe` invokes the same file), which regressed their previously
passing `run_strix_makefile.lit` with a Peano BFP kernel compile error. That
constraint is written up in #3459 for the remaining conversions.

Existing `Makefile`s and `run_makefile*.lit` are **untouched**, so Linux keeps its
current coverage while these examples gain a make-free path that also works on
Windows — nothing is orphaned.

## Validation

Green on Windows and both Linux Ryzen AI legs, with all 5 `run_cmake.lit` tests
building the design and running it on real NPU hardware through
`cmake -G Ninja` → `cmake --build` → `ctest` → `run_on_npu.py`.

**Note on an earlier revision of this PR.** The 85 → 90 pass-count delta reported
here previously did *not* mean the designs ran on hardware. `enable_testing()` was
being called inside `add_aie_run_test()`, where CMake does not emit
`CTestTestfile.cmake` — so every `add_test()` silently vanished, `ctest` printed
"No tests were found!!!", and it **exited 0**. The lits passed while the NPU was
never touched; the delta only reflected lit tests configuring and building.

Fixing that (`enable_testing()` at directory scope, plus `--no-tests=error` on
every lit so an empty suite fails loudly) immediately exposed a real bug:
`vector_scalar_add` loads its instruction stream with `xrt::elf`/`xrt::module`
and its Makefile runs with `-i ${elf_target}`, but the conversion fed it
`insts.bin`, so the host aborted with "is not a valid ELF file". It now passes
`ELF`/`USE_ELF`; the other four correctly read a raw `insts.bin`.

Also fixed here: `add_aie_design` attached the design build to `ALL`, but
`build_host_exe` configures the example's own `CMakeLists.txt`, so `make` JIT-ed
the design through CMake on top of its own `jit_xclbin` rule. It is now gated on
`AIE_BUILD_DESIGN`, which `build_host_exe` sets to `OFF` — this is what broke the
two `ml/block_datatypes` conversions mentioned above, and with the guard in place
they are re-landed in the follow-up.

## Outstanding (tracked in #3459)

- 20 host-C++ examples with a CMakeLists whose design parameters need threading
  through as CMake cache vars.
- 26 dirs without a CMakeLists (pure-Python designs + sweep dirs).
- 98 of the 126 make-gated lits are **parameter sweeps** (one Makefile, many env
  vars) — each parameterization needs its own configure dir; best done per family.
- Follow-on cleanup: reduce Makefiles to thin cmake-delegating wrappers, drop
  `makefile-common`'s `${powershell}`/`getwslpath`, broaden the Windows job.
- **Ceiling:** 14 Chess/Vitis lits can never run on native Windows →
  reachable parity is 112 of 126.

Refs #3453, #3459

### CODE REVIEW

**Comment by jgmelber** [comment]
### Validation status: CMake path proven on hardware ✅

CI is green (66 pass / 0 fail). Confirming the pilot actually exercised the new path, not just configured it:

On the **aie2-4col (npu1)** leg, `programming_examples` discovered **182** tests: 3 excluded, 92 unsupported, 1 failed, **85 passed**. `basic/vector_reduce_add/run_cmake.lit` appears in *neither* the Unsupported nor Excluded list, and the log doesn't print passing test names — so it ran and **passed**. That means `cmake -G Ninja` → `cmake --build` → `ctest` (which invokes `add_aie_run_test` → `run_on_npu.py`) successfully built the xclbin/insts and ran the design on real NPU hardware.

The one failure in the first attempt was `basic/matrix_scalar_add/run.lit` with `DRM_IOCTL_AMDXDNA_CREATE_HWCTX IOCTL failed (err=-2)` — the recurring XDNA driver flake on that runner, unrelated to this PR (that test is untouched here). It passed on rerun.

Windows and aie2p-8col also pass; `run_cmake.lit` is correctly Unsupported on npu2 since it's `ryzen_ai_npu1`-gated.

**So the pattern is validated** and the follow-up sweep (converting the remaining make-only examples onto `add_aie_design`/`add_aie_run_test`, reducing Makefiles to thin cmake-delegating wrappers, and broadening the Windows gate) can proceed on this foundation.

File: python/aie_lit_utils/lit_config_helpers.py [hunk 1]
```diff
@@ -110,6 +110,16 @@ def add_makefile_examples_feature(config_obj) -> None:
...
 
+    @staticmethod
+    def add_cmake_examples_feature(config_obj) -> None:
+        """Enable CMake-based example tests when cmake is available.
+
+        Unlike the Make path, this works on Windows too — it is how example
+        designs build+run without GNU make (see add_aie_design/add_aie_run_test
+        in programming_examples/common.cmake)."""
+        if shutil.which("cmake"):
+            config_obj.available_features.add("cmake_examples")
```
[the remark this answers is not in the document]
Comment (jgmelber) [reply python/aie_lit_utils/lit_config_helpers.py]: Good catch — fixed in add32fa. The feature now requires `cmake`, `ctest` **and** `ninja`, matching what `run_cmake.lit` actually invokes (`cmake -G Ninja`, `cmake --build`, `ctest`), so a host with cmake but no ninja/ctest marks these Unsupported rather than selecting and failing them.

File: programming_examples/common.cmake [hunk 2]
```diff
@@ -205,3 +205,84 @@ function(target_link_test_utils target_name)
...
+# -----------------------------------------------------------------------------
+# Make-free NPU design build + run helpers
+# -----------------------------------------------------------------------------
+# These let an example build its xclbin/insts and run on the NPU entirely
+# through cmake + ctest, with no GNU make and no shell (the process is launched
+# directly, so no powershell/wslpath wrapping is needed on any host). They are
+# the CMake equivalents of makefile-common's jit_xclbin (design build) and the
+# per-example `run:` target.
+
+find_package(Python3 COMPONENTS Interpreter REQUIRED)
```
[the remark this answers is not in the document]
Comment (jgmelber) [reply programming_examples/common.cmake]: Agreed, this was a real regression — fixed in add32fa. `main` only probed for Python `QUIET` inside the repo-root fallback, so my unconditional file-scope `REQUIRED` would have forced an interpreter on every consumer of `common.cmake`, including host-only builds. The requirement now lives in a small `_aie_require_python()` helper called from `add_aie_design` and `add_aie_run_test` — the only places that actually use it — and it no-ops if Python was already found.

File: programming_examples/lit.cfg.py [hunk 3]
```diff
@@ -34,6 +34,7 @@
 )
 
 LitConfigHelper.add_makefile_examples_feature(config)
+LitConfigHelper.add_cmake_examples_feature(config)
```
[the remark this answers is not in the document]
Comment (jgmelber) [reply programming_examples/lit.cfg.py:37]: Correct, the description was stale — updated. The branch has **5** converted examples. The two `ml/block_datatypes` ones were converted and then reverted: appending the helpers to their `CMakeLists.txt` also changed the *make* build (`build_host_exe` invokes the same file), which regressed their previously-passing `run_strix_makefile.lit` with a Peano BFP kernel compile error (`bfp16ebs8` / `block_vector` / `v128bfp16ebs8` unknown). That constraint is documented in #3459 so the remaining conversions avoid it.

**Review by hunhoffe — APPROVED** [review]

**Comment by jgmelber** [comment]
Pushed two commits (`a32a726`, `61b6e47`) fixing defects found while building the follow-up (#3468). Re-review needed — flagging since this was previously approved and queued.

**The converted tests were passing without running.** `enable_testing()` was called inside `add_aie_run_test()`, where CMake does not write `CTestTestfile.cmake`. Every `add_test()` silently vanished, `ctest` reported "No tests were found!!!" — and exited **0**, so `run_cmake.lit` passed. All five examples were configuring and building but never touching the NPU. The 85 → 90 delta in the description reflected lit tests, not hardware runs; I've corrected that section.

Fixed by moving `enable_testing()` to directory scope and adding `--no-tests=error` to each lit so an empty suite fails loudly instead of passing.

**That immediately exposed a real bug.** With the tests actually running, CI failed on `basic/vector_scalar_add/run_cmake.lit`:

```
what():  .../insts.bin is not found or is not a valid ELF file
```

`vector_scalar_add` loads its instruction stream via `xrt::elf` + `xrt::module`, and its Makefile uses `jit_xclbin_elf` with `-i ${elf_target}` — but the conversion passed `insts.bin`. So the PR's flagship example was broken on the CMake path from the start, concealed by its own passing test. It now passes `ELF`/`USE_ELF`. I audited the other four: all read a raw `insts.bin` and correctly omit it.

**Two supporting changes.** `TARGET_NAME` defaulted to `test`, which CTest reserves once testing actually works, so every example failed to configure — now `test_exe` (every real invocation passes `-DTARGET_NAME` anyway). And `add_aie_design` is now gated on `AIE_BUILD_DESIGN`, which `build_host_exe` sets to `OFF`: attaching the design to `ALL` meant `make` JIT-ed it through CMake on top of its own `jit_xclbin` rule. That is what broke the two `ml/block_datatypes` conversions reverted here; with the guard they are re-landed in #3468 and **passed on hardware**.

Verified locally that all five examples register exactly one ctest, `vector_scalar_add` gets `-i final.elf` while the rest get `insts.bin`, and `-DAIE_BUILD_DESIGN=OFF` emits zero JIT commands while still building the host exe. The ELF fix is already confirmed green on the Linux `aie2-4col` leg in #3468.
